### ISSUE TITLE
v5 post requests does not include the body 

### ISSUE THREAD

**Issue by NikosEfthias** [sagebind/chttp#51]
I had a small twilio api binding in my program which worked fine until i upgraded the version to v5.
After the upgrade i start to receive missing body from twilio. I today downgraded and everything again works fine. You can see below how i use the function. changing nothing and upgrading to `v5` breaks the code downgrading to v4 everything runs cool again 
```rust
chttp::http::Request::post(url)
	.header(
		"Authorization",
			format!(
				"Basic {}",
					encode(format!("{}:{}", TWILIO_SID, TWILIO_TOKEN).as_str())
			),
	)
	.header("Content-Type", "application/x-www-form-urlencoded")
	.body(body),
```

**Comment by sagebind** [comment]
Thanks for filing this issue, I will look into this shortly!

I have a high suspicion as to the cause of this. libcurl is rather picky on whether or not it sends a body provided based on how the headers and request method are set... I will write up a test case for this.

**Comment by sagebind** [comment]
This has been fixed in the 0.5.4 release!

**Comment by NikosEfthias** [comment]
i will test this shortly today i really appreciate the effort you put in this library
